src/idl_gen_cpp.cpp

```diff
@@ -2273,11 +2273,20 @@ class CppGenerator : public BaseGenerator {
   void GenKeyFieldMethods(const FieldDef &field) {
     FLATBUFFERS_ASSERT(field.key);
     const bool is_string = (IsString(field.value.type));
+    const bool is_array = (IsArray(field.value.type));
 
     code_ += "  bool KeyCompareLessThan(const {{STRUCT_NAME}} *o) const {";
     if (is_string) {
       // use operator< of flatbuffers::String
       code_ += "    return *{{FIELD_NAME}}() < *o->{{FIELD_NAME}}();";
+    } else if (is_array){
```

Comment:
Did you run the scripts/git-clang-format script? Seems like there should be a space after the parentheses. I won't comment on other formatting issues at the moment.